packages/compass-saved-aggregations-queries/src/components/edit-item-modal.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Modal,
+  H3,
+  TextInput,
+  spacing,
+  Button,
+  css,
+} from '@mongodb-js/compass-components';
+import { connect } from 'react-redux';
+import type { MapDispatchToProps, MapStateToProps } from 'react-redux';
+import type { RootState } from '../stores';
+import type { UpdateItemAttributes } from '../stores/edit-item';
+import type { Item } from '../stores/aggregations-queries-items';
+import { cancelEditItem, updateItem } from '../stores/edit-item';
+
+type EditItemModalProps = {
+  isModalOpen: boolean;
+  item?: Pick<Item, 'id' | 'name' | 'type'>;
+  onSubmit(id: string, attributes: UpdateItemAttributes): void;
+  onCancel: () => void;
+};
+
+const formTitleStyles = css({
+  marginBottom: spacing[5],
+});
+
+const formFooterStyles = css({
+  marginTop: spacing[5],
+  display: 'flex',
+  justifyContent: 'flex-end',
+});
+
+const cancelButtonStyles = css({
+  marginRight: spacing[2],
+});
+
+const EditItemModal: React.FunctionComponent<EditItemModalProps> = ({
+  isModalOpen,
+  item,
+  onSubmit,
+  onCancel,
+}) => {
+  const [name, setName] = useState(item?.name ?? '');
+  useEffect(() => {
+    setName(item?.name ?? '');
+  }, [item]);
+
+  if (!item) {
+    return <></>;
```

Comment:
I think a more conventional way of handling this is to return `null` here and not a fragment. IIRC the reason for this was that `null`s were handled as a sorta special empty slot case in React reconciler (there is a [reference](https://reactjs.org/docs/implementation-notes.html#what-we-left-out) to this in the docs, but not much detail). It's probably not important in that particular case, just FYI